package/src/components/ChannelPreview/hooks/__tests__/useChannelPreviewDisplayName.test.js

```diff
@@ -0,0 +1,139 @@
+import { useMockedApis } from '../../../../mock-builders/api/useMockedApis';
+import { generateChannelResponse } from '../../../../mock-builders/generator/channel';
+import { generateUser } from '../../../../mock-builders/generator/user';
+import { getTestClientWithUser } from '../../../../mock-builders/mock';
+import {
+  getChannelPreviewDisplayName,
+  useChannelPreviewDisplayName,
+} from '../useChannelPreviewDisplayName';
+
+describe('useChannelPreviewDisplayName', () => {
+  const clientUser = generateUser();
+  let chatClient;
+  let channel;
+  const initializeChannel = async (c) => {
+    useMockedApis(chatClient, [getOrCreateChannelApi(c)]);
+
+    channel = chatClient.channel('messaging');
+
+    await channel.watch();
+  };
+
+  beforeEach(async () => {
+    chatClient = await getTestClientWithUser(clientUser);
+  });
+
+  afterEach(() => {
+    channel = null;
+  });
+
+  it('will return a channel display name', async () => {
+    const channelName = 'okechukwu';
+    const characterLength = 15;
+    await initializeChannel(
+      generateChannelResponse({
+        channel: {
+          name: channelName,
+        },
+      }),
+    );
+
+    const TestComponent = () => {
+      const channelDisplayName = useChannelPreviewDisplayName(channel, characterLength);
+
+      return <Text>{channelDisplayName}</Text>;
+    };
+
+    const { getByText } = render(<TestComponent />);
+
+    await waitFor(() => {
+      expect(getByText(channelName)).toBeTruthy();
+    });
+  });
+
+  it('will return the same channelName when channelName length is less than characterLength', async () => {
+    const channelName = 'okechukwu';
+    const characterLength = 15;
+    const currentUserId = chatClient.userID;
+    await initializeChannel(
+      generateChannelResponse({
+        channel: {
+          name: channelName,
+        },
+      }),
+    );
+
+    const TestComponent = () => {
+      const channelDisplayName = useChannelPreviewDisplayName(channel, characterLength);
+
+      return <Text>{channelDisplayName}</Text>;
+    };
+
+    render(<TestComponent />);
+
+    const displayName = getChannelPreviewDisplayName({
+      channelName,
+      currentUserId,
+      maxCharacterLength: characterLength,
+      members: channel['state']['members'],
+    });
+
+    expect(displayName).toEqual(channelName);
+  });
+
+  it('will return the same channelName length as the characterLength', async () => {
+    const channelName = 'okechukwu nwagba martin';
+    const characterLength = 15;
+    const currentUserId = chatClient.userID;
+    await initializeChannel(
+      generateChannelResponse({
+        channel: {
+          name: channelName,
+        },
+      }),
+    );
+
+    const displayName = getChannelPreviewDisplayName({
+      channelName,
+      currentUserId,
+      maxCharacterLength: characterLength,
+      members: channel['state']['members'],
+    });
+
+    expect(displayName.length).toEqual(channelName.length);
```

Comment:
It might be even more explicit to just ` expect(displayName).toEqual(channelName);` you'll get a full equality